.github/workflows/ci-crossdock.yml

```diff
@@ -0,0 +1,70 @@
+name: Crossdock
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  crossdock:
+    runs-on: ubuntu-latest
+    name: Crossdock
+    steps:
+    - uses: actions/checkout@v2
+    
+    - uses: actions/setup-java@v2
+      with:
+        java-version: "1.8"
+    
+    - uses: docker/login-action@v1
+      id: dockerhub-login
+      with:
+        username: jaegertracingbot
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+      env:
+        DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+      if: env.DOCKERHUB_TOKEN != null
+
+    - name: Export DOCKERHUB_LOGIN variable
+      run: echo "DOCKERHUB_LOGIN=true" >> $GITHUB_ENV
+      if: steps.dockerhub-login.outcome == 'success'
+
+    - name: Export Branch variable for pull_request event
+      id: branch-pr
+      run: |
+        export BRANCH=${GITHUB_HEAD_REF}
+        echo "Running on branch=$BRANCH"
+        echo "BRANCH=${BRANCH}" >> $GITHUB_ENV
+        echo ::set-output name=branch::$BRANCH
+      if: github.event_name == 'pull_request'
+    
+    - name: Export Branch variable for push event
+      id: branch-push
+      run: |
+        export BRANCH=${GITHUB_REF##*/}
+        echo "Running on branch=$BRANCH"
+        echo $BRANCH >> $GITHUB_ENV
+        echo ::set-output name=branch::$BRANCH
+      if: github.event_name == 'push'
+
+    - name: Run Crossdock
+      id: crossdock
+      run: make crossdock-fresh
+
+    - name: Publish crossdock
+      run: |
+        REPO=jaegertracing/xdock-java
+        TAG=$(if [ "$BRANCH" == "master" ]; then echo "latest"; else echo "${BRANCH///}"; fi)
+        COMMIT=${GITHUB_SHA::8}
+        echo "BRANCH=$BRANCH, REPO=$REPO, TAG=$TAG, COMMIT=$COMMIT"
+        ./gradlew shadowJar
+        pushd 'jaeger-crossdock'
+        docker build -f Dockerfile -t $REPO:$COMMIT
```

Comment:
please check this PR https://github.com/jaegertracing/jaeger-client-node/pull/454 on how to avoid dual-build of the Docker image and reuse the one from `make crossdock-fresh`

Reply:
@yurishkuro If I'm reading that change right, I just need to add `image:` in the docker-compose file right? The difference between the node and the java example here is that I see two java's in the docker-compose file in this repo (`java-http` and `java-udp`) should it be one of those? Or should there be a seperate `java` declaration to do the build? Looks like the only difference between the two existing is the environment variable, so not sure how much that matters when it would be published.

Reply:
> If I'm reading that change right, I just need to add image: in the docker-compose file right?

I think so, and remove the 2nd build step.

> (java-http and java-udp) should it be one of those?

Yes, I think the UDP one is the "official" image that will be used by crossdock tests in other Jaeger SDKs. You can confirm it by comparing with what is being built in the 2nd build step.

Reply:
@yurishkuro Sorry it's taken a bit of time to get back to this. The only difference I see between the UDP and Dockerfile is that there is an environment variable being set (`sender`). Looks like that var is different for the http and udp variants, so I'm assuming that it has some effect in the code. Would it be better to  just add a "plain" java to the docker-compose file and then use that to push? Or if the var has no real impact, then we can just go ahead and use the UDP one that is already there.

Reply:
UDP is the default mode, if the `SENDER` env var is not specified it will default to UDP. 

https://github.com/jaegertracing/jaeger-client-java/blob/2be525bdcf7021b4d250f95ca4b6e118a684da4c/jaeger-crossdock/src/main/java/io/jaegertracing/crossdock/JerseyServer.java#L149-L152

So `java-udp` from docker-compose is the image we want to publish.

Reply:
Got it. This should be resolved now.